conda_vendor/conda_vendor.py

```diff
@@ -1,218 +1,499 @@
-        if entry['channel'].startswith(chan):
+    for entry in solution["actions"]["FETCH"]:
+        if entry["channel"].startswith(channel):
             continue
         fetch.append(entry)
 
-    for entry in dry_run_install['actions']['LINK']:
-        if entry['base_url'] == chan:
+    for entry in solution["actions"]["LINK"]:
+        if entry["base_url"] == channel:
             continue
         link.append(entry)
 
-    dry_run_install['actions']['FETCH'] = fetch
-    dry_run_install['actions']['LINK'] = link
-    return dry_run_install
+    solution["actions"]["FETCH"] = fetch
+    solution["actions"]["LINK"] = link
+    return solution
 
-def solve_environment(lock_spec, solver, platform) -> DryRunInstall:
-    specs = get_specs(lock_spec)
 
-    click.echo(click.style(f"Using Solver: {solver}", bold=True, bg='black', fg='cyan'))
-    click.echo(click.style(f"Solving for Platform: {platform}", bold=True, bg='black', fg='cyan'))
-    click.echo(click.style(f"Solving for Spec: {specs}", bold=True, bg='black', fg='cyan'))
+def solve_environment(
+    environment_file: Path, solver: str, platform: str
+) -> List[FetchAction]:
+    """Solve the environment specified in the conda environment_yaml,
+    and return a list of all the required packages with enough metadata
+    to generate a repo_data.json.
 
-    virtual_package_repodata = CondaLockWrapper.default_virtual_package_repodata()
-    virtual_package_chan = virtual_package_repodata.channel
-    channels = [*lock_spec.channels, virtual_package_chan]
-    dry_run_install = CondaLockWrapper.solve_specs_for_arch(
-            solver,
-            channels,
-            specs,
-            platform)
+    Parameters
+    ----------
+    environment_file: pathlib.Path
+        path to the environment.yaml to solve
 
-    dry_run_install = _scrub_virtual_pkgs(dry_run_install,
-                                          virtual_package_chan.url)
+    solver: str
+        which conda is used to solve: conda, miniconda, mamba, etc.
 
-    if not dry_run_install['success']:
-        sys.exit("Failed to Solve for {specs}\n Using {solver} for {platform}")
-    click.echo(click.style("Successfull Solve", bold=True, fg='green', blink=True))
+    platform: str
+        platform to vendor for: eg. linux-32, win-64, osx-64, etc.
 
-    return dry_run_install
+    Returns
+    -------
+    list [ conda_lock.conda_solver.FetchAction ]
 
+    """
+    assert isinstance(environment_file, Path)
 
-# get formatted List(str) to pass to CondaLockWrapper.solve_specs_for_arch()
-def get_specs(lock_spec) -> List[str]:
-    versioned_deps = lock_spec.dependencies
-    specs = []
-    for dep in versioned_deps:
-        if dep.version == '':
-            specs.append(f"{dep.name}")
-        else:
-            specs.append(f"{dep.name}=={dep.version}")
-    return specs
+    # generate conda-lock's LockSpecification
+    lock_spec = parse_environment_file(environment_file)
```

Comment:
The signature of `conda_lock.src_parser.parse_environment_file` changed from 1.2.1 to 1.3, there is now an additional parameter `given_platforms : Optional[Sequence[str]]`. Either pin the conda_lock version to 1.2.1 or fix the invocation of this function.

Reply:
should be fixed.